### PR TITLE
Created a article under foundations of when to NOT use agents

### PR DESCRIPTION
**Pull request by onlywmo**
## Summary

Describe the change in 2-5 sentences. Keep the scope concrete.

## Target Branch

- Normal contributor work targets `develop`.
- Release PRs target `main` only when opened by `dprat0821` from same-repository `develop`.
- `main` is the production Mintlify branch and should not receive direct feature PRs.

## Contribution Type

Select the best match and remove the rest.

- lab article
- major revision to existing page
- radar note
- source project
- curated reference note
- repo/process/meta

## Placement

- Target path:
- Links or indexes updated:
- Related issue:

## Source Lineage

List the outside sources, imported references, or existing repo pages that shaped this change.

## Review Checklist

- [ ] The contribution type is explicit.
- [ ] The file or folder is in the correct path for that artifact type.
- [ ] I followed the matching template or contributor guidance where relevant.
- [ ] The change stays repo-native and does not copy upstream material into public tracked paths.
- [ ] Citations, source notes, or attribution boundaries are clear where needed.
- [ ] Relevant README, reading-path, or contributor links were added or updated for discoverability.
- [ ] If I added or revised a practitioner skill package, it includes a human-facing `README.md` and an agent-facing `SKILL.md`.
- [ ] Status and maintenance expectations are clear.
- [ ] The linked issue has the correct `track:*` and `kind:*` labels, and this PR only changes paths allowed for that track.
- [ ] If I changed example code, I ran `python3 scripts/verify_example_projects.py`.
- [ ] If I renamed files or changed paths, I ran `python3 scripts/check_filename_casing.py`.

### CODE REVIEW

**Review by gongxiaojing0825 — CHANGES_REQUESTED** [review]
hi @onlywmo , you should make PR towards /develop branch rather than to /main

File: foundations/when-NOT-to-use-agents.mdx [hunk 1]
```diff
@@ -0,0 +1,108 @@
+---
+title: LLM Foundations For Agent Systems
+owner: Prompthon IO
+updated: 2026-04-21
+depth: foundational
+region_tags:
+  - global
+coding_required: optional
+external_readings: []
```
Comment (dprat0821) [inline foundations/when-NOT-to-use-agents.mdx]: Please repair the front matter before re-review. The title is copied from another page, the `updated` date predates this article, and `external_readings` is empty even though this draft derives a decision framework from established industry discussion. Use the actual title and revision date, then record the main first-party sources. The handbook’s article guidelines require durable articles to cite their main sources.

File: foundations/when-NOT-to-use-agents.mdx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  - **Over-engineered solution** that are harder to maintain than necessary
+  - **Unpredictable Cost** from multi-step LLM calls that could've been avoided
+  - **Unreliable behavior** in scenarios where determinism is required 
+  - **Team friction** when agents are introduced into contexts that don't actually benefit from autonomy 
+
+Choosing an agents when a simpler approach would work is often more damaging than choosing the wrong model. 
+
+## Mental Model
+
+The cleanest distinction between using an agent or using a simpler tool is about whether the task requires runtime adaptation.
```
Comment (dprat0821) [inline foundations/when-NOT-to-use-agents.mdx]: This is useful, but it is established industry framing rather than a new standalone distinction. [Anthropic](https://www.anthropic.com/engineering/building-effective-agents) distinguishes predefined-code workflows from agents that dynamically direct their process and tool use; [Microsoft](https://learn.microsoft.com/en-us/agent-framework/journey/workflows) frames the decision as “who should decide what happens next?” Please cite and synthesize that prior discussion, and explain what this page adds beyond the handbook’s existing `/foundations/agents-vs-workflows` page.

File: foundations/when-NOT-to-use-agents.mdx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+| Dimension | Use a Simpler Approach | Consider an Agent |
+|---|---|---|
+| **Path** | Fully known in advance | Unknown or changes dynamically |
+| **Rules** | Stable and predictable | Ambiguous or context-dependent |
+| **Cost tolerance** | Low (budget-conscious) | Higher (value justifies cost) |
+| **Latency tolerance** | Low (real-time needed) | Higher (seconds/minutes acceptable) |
+| **Accuracy requirement** | 100% (deterministic) | Tolerates some variability |
```
Comment (dprat0821) [inline foundations/when-NOT-to-use-agents.mdx]: This conflates accuracy with determinism. Deterministic code can be consistently wrong, while an agentic step can be tested, validated, or held for approval before its output is used. For high-stakes work, current guidance emphasizes bounded autonomy, deterministic validation, and human oversight—not a categorical ban on agents. Reframe this dimension as something like “tolerance for unreviewed model error / required verification and control,” and distinguish agent-assisted analysis from the final authoritative calculation or action. See [OpenAI’s high-risk-action guidance](https://openai.com/business/guides-and-resources/a-practical-guide-to-building-ai-agents/) and [Google Cloud’s human-in-the-loop pattern](https://docs.cloud.google.com/architecture/choose-design-pattern-agentic-ai-system).

File: foundations/when-NOT-to-use-agents.mdx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+**2. Deterministic Business Logic**
+  - Example: Monthly invoice generation
+
+**3. Real-time or Low-latency applications** 
+  - Example: A search bar that suggests results as the user types. Agents would be too slow
+
+**4. Cost-sensitive environments**
+
+  - Example: A high-volume data processing pipelines processing 10M records daily. Agents would be financially unsustainable.
```
Comment (dprat0821) [inline foundations/when-NOT-to-use-agents.mdx]: “Financially unsustainable” cannot be concluded from record count alone. Unit cost depends on the model, tokens, calls or turns, batching or caching, retries, and the value of a successful outcome. [Anthropic](https://www.anthropic.com/engineering/building-effective-agents) describes cost and latency as tradeoffs that must be justified; [Google Cloud](https://docs.cloud.google.com/architecture/choose-design-pattern-agentic-ai-system) likewise treats them as workload constraints. Please make this conditional and recommend measuring cost per completed task, retry rate, and p95 latency instead of asserting an absolute.

File: foundations/when-NOT-to-use-agents.mdx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+**5. High-stakes/100% Accuracy Scenarios**
+  - Example: Calculating tax withholdings. There is no room for an agent to "decide" the wrong number.
+
+**6. No clear evaluation metrics**
+  - Example: A creative writing assistant where "quality" is entirely subjective.
```
Comment (dprat0821) [inline foundations/when-NOT-to-use-agents.mdx]: Subjective quality is not the same as “no evaluation.” Creative and research outputs can be evaluated with explicit rubrics, pairwise comparisons, model-based graders calibrated against humans, A/B tests, and expert review. [Anthropic’s current eval guidance](https://www.anthropic.com/engineering/demystifying-evals-for-ai-agents) explicitly covers open-ended and subjective tasks. The safer claim is: if the team cannot define success criteria and an evaluation process, the system is not ready to launch—not that an agent is categorically unsuitable.

File: foundations/when-NOT-to-use-agents.mdx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+  - Example: A creative writing assistant where "quality" is entirely subjective. 
+
+**7. Limited engineering resources**
+  - Example: A small startup's first AI feature. Start with a prompt; add an agent later if needed.
+
+## Architecture Diagram 
+
+```mermaid
+flowchart TD
+    Start["Start: Is there a task?"] --> Q1{"Can a single prompt<br>or simple RAG solve it?"}
```
Comment (dprat0821) [inline foundations/when-NOT-to-use-agents.mdx]: RAG is not mutually exclusive with agents; it is a retrieval and context technique that can support a single model call, a deterministic workflow, or an agent. Microsoft’s [workflow spectrum](https://learn.microsoft.com/en-us/agent-framework/journey/workflows) explicitly places RAG among context providers and shows that most production systems are hybrids. Consider: “Can a single model call, optionally augmented with retrieval, meet the measured requirements?” Then let later branches compare deterministic, hybrid, and agent-directed control.

File: foundations/when-NOT-to-use-agents.mdx [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    Q5 -->|YES| Stop5["DON'T USE AGENT"]
+    Q5 -->|NO| Consider["CONSIDER USING AGENT"]
+```
+
+## Tradeoffs
+
+The choice between using and not using an agent involves four core tradeoffs: 
+  - **Simplicity VS Adaptability**: Simpler solutions are easier to build and maintain, but agents adapt better to changing contexts. Workflows are easier to test and audit, but they are expensive to maintain when edge cases keep growing. 
+  - **Cost VS Capability**: The added capability must justify the added expense.
+  - **Speed VS Intelligence**: Agents are smarter but slower. Trade intelligence for speed if speed is critical in your situation.
```
Comment (dprat0821) [inline foundations/when-NOT-to-use-agents.mdx]: “Agents are smarter” is technically misleading. An agent is a control and orchestration pattern; capability comes from the model, tools, context, instructions, and evaluation harness. Iteration and tool use may improve task performance, but they can also compound errors. Reframe this as **Latency and cost vs. task performance or adaptability**, matching [Anthropic’s stated tradeoff](https://www.anthropic.com/engineering/building-effective-agents) and [OpenAI’s model-selection guidance](https://openai.com/business/guides-and-resources/a-practical-guide-to-building-ai-agents/).

File: foundations/when-NOT-to-use-agents.mdx [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+  - **Predictability VS Flexibility**: Deterministic code is auditable and predictable. Agents are more flexible but harder to test and certify.
+
+**Useful Defaults:**
+  - Default to simple prompts and scripts for deterministic business policy, audit-heavy paths, and anything that requires 100% accuracy. 
+  - Default to agents only for bounded exploration and judgement--and only after you've proven that simpler approaches can't solve the problem. 
+  - When in doubt, start simple. Build a basic prompt or workflow first, then monitor it in production. Only add agentic production when you have clear evidence that the problem geniunely requires runtime adaptation. 
+
+## Reading Extensions
+
+- [Evaluation and Observability](systems/evaluation-and-observability)
```
Comment (dprat0821) [inline foundations/when-NOT-to-use-agents.mdx]: This relative link will resolve under the current `/foundations/` route, producing `/foundations/systems/evaluation-and-observability` instead of the intended page. Please use `/systems/evaluation-and-observability` and verify all internal links in the local docs preview before requesting re-review.

**Review by dprat0821 — CHANGES_REQUESTED** [review]
This is a valuable topic and it fits the gap approved in #200, but the current draft is not ready as a durable handbook page. The central recommendation—start simple and add model-directed control only when the task warrants it—is broadly aligned with current practice; several supporting statements, however, are presented as universal facts when the industry guidance is more nuanced.

Before requesting re-review, please:

1. Retarget or reopen the contribution against `develop`, as the earlier review notes. The current `main` target violates the contribution flow and the track-guard check is failing.
2. Replace the untouched PR template with a concrete summary, placement, `Closes #200`, source lineage, and completed checks.
3. Complete the integration promised in #200: add the page to `foundations/index.mdx`, `foundations/README.md`, and the Foundations navigation in `docs.json`. This PR currently adds only the article file.
4. Follow the article contract: add the missing **Tool Landscape** and **Citations** sections, correct the metadata, and rename the slug to lowercase (`foundations/when-not-to-use-agents.mdx`).
5. Revise the factual and architectural claims called out inline, then do a full copyedit and verify every internal link in a local preview.

Please read and cite these first-party sources before the next revision:

- [Anthropic — Building effective agents](https://www.anthropic.com/engineering/building-effective-agents)
- [OpenAI — A practical guide to building agents](https://openai.com/business/guides-and-resources/a-practical-guide-to-building-ai-agents/)
- [Microsoft — Workflows in the Agent Development Journey](https://learn.microsoft.com/en-us/agent-framework/journey/workflows)
- [Google Cloud — Choose a design pattern for your agentic AI system](https://docs.cloud.google.com/architecture/choose-design-pattern-agentic-ai-system)
- [Anthropic — Demystifying evals for AI agents](https://www.anthropic.com/engineering/demystifying-evals-for-ai-agents)

A stronger revision should present a spectrum—single model call (optionally with retrieval), deterministic workflow, hybrid workflow with bounded agentic steps, and model-directed agent loop—and explain where controls, evaluation, human approval, cost, and latency change along that spectrum.

**Comment by dprat0821** [comment]
should be PR to dev branch.
